internal/atom/kubernetes/engine.go

```diff
@@ -0,0 +1,156 @@
+package kubernetes
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os/user"
+	"path/filepath"
+
+	"github.com/caesium-dev/caesium/internal/atom"
+	"github.com/caesium-dev/caesium/pkg/env"
+	"github.com/google/uuid"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// Engine defines the interface for treating the
+// Kubernetes API as a atom.Engine.
+type Engine interface {
+	atom.Engine
+}
+
+type kubernetesEngine struct {
+	ctx     context.Context
+	backend kubernetesBackend
+}
+
+// NewEngine creates a new instance of kubernetes.Engine
+// for interacting with kubernetes.Atoms.
+func NewEngine(ctx context.Context) Engine {
+	k8sCfg := env.Variables().KubernetesConfig
+	if k8sCfg == "" {
+		u, _ := user.Current()
+		k8sCfg = filepath.Join(u.HomeDir, kubeConfig)
+	} else {
+		k8sCfg = filepath.Join(k8sCfg, kubeConfig)
+	}
+
```

Comment:
Where is `kubeConfig` coming from?

Reply:
It's a constant in another file. I need to still refactor this a bit so the unit tests work if there is no ~/.kube/config file. They ran on my machine but not in CircleCI.